### PR TITLE
Add an upper bound for the video decoder workaround

### PR DESCRIPTION
**Pull request by JanuszL**
- there is a bug when decoding video on non-default stream
  in selected driver leading to memory corruption. This
  PR adds a check for the last driver version that needs
  the workaround

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [X] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR
- there is a bug when decoding video on non-default stream
  in selected driver leading to memory corruption. This
  PR adds a check for the last driver version that needs
  the workaround

#### Additional information
- Affected modules and functionalities:
  - video loader

- Key points relevant for the review:
  - NA

### Tests
- [X] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [X] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [X] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2534

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: dali/operators/reader/nvdecoder/nvdecoder.cc [hunk 1]
```diff
@@ -59,22 +59,21 @@ NvDecoder::NvDecoder(int device_id,
...
+  // This is a workaround for an issue with nvcuvid in drivers >460 and <= 470.21 where concurrent
   // use on default context and non-default streams may lead to memory corruption.
-  // TODO(michalz): add an upper bound when the problem is fixed
   bool use_default_stream = false;
 #if NVML_ENABLED
   {
     nvml::Init();
     static float driver_version = nvml::GetDriverVersion();
-    if (driver_version > 460)
+    if (driver_version > 460 && driver_version <= 470.21)
```
Comment (mzient) [inline dali/operators/reader/nvdecoder/nvdecoder.cc:69]:
```suggestion
    if (driver_version > 460 && driver_version < 470.21)
```

Comment (JanuszL) [reply dali/operators/reader/nvdecoder/nvdecoder.cc]: Done

**Comment by JanuszL** [comment]
!build

**Review by mzient — APPROVED** [review]

**Pull request merged**